### PR TITLE
graph: backend: elyzor: verify gc API version when loading '.so'

### PR DESCRIPTION
**Pull request by dchigarev**
What this PR does:
1. Adds the version getter (`dnnl_graph_compiler_get_version`) to the vtable `dnnl_graph_compiler_vtable`
2. During `libgraph_compiler.so` loading, verifies that the API version in the `.so` matches the one defined by oneDNN

### CODE REVIEW

File: src/graph/backend/elyzor/utils.cpp [hunk 1]
```diff
@@ -95,6 +96,23 @@ std::vector<op_kind_t> get_supported_op_kinds() {
...
+std::string gc_version_to_string(
+        const dnnl_graph_compiler_version::version &v) {
+    std::stringstream ss;
+    // Cast uint8_t to int before inserting into the stringstream
+    // as otherwise they're interpreted as characters
+    ss << static_cast<int>(v.major) << "." << static_cast<int>(v.minor) << "."
+       << static_cast<int>(v.patch) << " " << v.hash;
+    return ss.str();
+}
+
```
Comment (AndreyPavlenko) [inline src/graph/backend/elyzor/utils.cpp]: Currently the `gc_version_to_string()` functions are used only to build the error message.  `std::stringstream` is created 3 times to build a string. You could avoid building the strings by passing the `std::stringstream` to these functions and printing into the same stream.

File: src/graph/backend/elyzor/utils.hpp [hunk 2]
```diff
@@ -78,4 +79,10 @@ std::vector<op_kind_t> get_supported_op_kinds();
 } // namespace graph
 } // namespace impl
 } // namespace dnnl
+
```
Comment (dchigarev) [inline src/graph/backend/elyzor/utils.hpp:82]: declaring the operators outside of namespaces so they would be available everywhere

File: src/graph/backend/elyzor/utils.cpp (hunk 1 above)
Comment (dchigarev) [reply src/graph/backend/elyzor/utils.cpp]: done, defined `operator<<` for versions instead

Comment (AndreyPavlenko) [reply src/graph/backend/elyzor/utils.cpp]: This is even better.

**Review by AndreyPavlenko — APPROVED** [review]

**Review by kurapov-peter — APPROVED** [review]

**Pull request merged**